docs/modules/rust/index.md

```diff
@@ -1,514 +1,7 @@
-
-Our `Person` table has a unique id column, so we can filter for a row matching that ID. Since it is unique, we will find either 0 or 1 matching rows in the database. This gets represented naturally as an `Option<Person>` in Rust. SpacetimeDB automatically creates and uses indexes for filtering on unique columns, so it is very efficient.
-
-The name of the filter method just corresponds to the column name.
-
-```rust
-#[reducer]
-fn filtering(ctx: &ReducerContext, id: u64) {
-    match ctx.db.person().id().find(id) {
-        Some(person) => println!("Found {person}"),
-        None => println!("No person with id {id}"),
-    }
-}
-```
-
-Our `Person` table also has an index on its `age` column. Unlike IDs, ages aren't unique. Filtering for every person who is 21, then, gives us an `Iterator<Item = Person>` rather than an `Option<Person>`.
-
-```rust
-#[reducer]
-fn filtering_non_unique(ctx: &ReducerContext) {
-    for person in ctx.db.person().age().filter(21u32) {
-        println!("{} has turned 21", person.name);
-    }
-}
-```
-
-> NOTE: An unfortunate interaction between Rust's trait solver and integer literal defaulting rules means that you must specify the types of integer literals passed to `filter` and `find` methods via the suffix syntax, like `21u32`. If you don't, you'll see a compiler error like:
-> ```
-> error[E0271]: type mismatch resolving `<i32 as FilterableValue>::Column == u32`
->    --> modules/rust-wasm-test/src/lib.rs:356:48
->     |
-> 356 |     for person in ctx.db.person().age().filter(21) {
->     |                                         ------ ^^ expected `u32`, found `i32`
->     |                                         |
->     |                                         required by a bound introduced by this call
->     |
->     = note: required for `i32` to implement `BTreeIndexBounds<(u32,), SingleBound>`
-> note: required by a bound in `BTreeIndex::<Tbl, IndexType, Idx>::filter`
->     |
-> 410 |     pub fn filter<B, K>(&self, b: B) -> impl Iterator<Item = Tbl::Row>
->     |            ------ required by a bound in this associated function
-> 411 |     where
-> 412 |         B: BTreeIndexBounds<IndexType, K>,
->     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `BTreeIndex::<Tbl, IndexType, Idx>::filter`
-> ```
-
-### Deleting
-
-Like filtering, we can delete by an indexed or unique column instead of the entire row.
-
-```rust
-#[reducer]
-fn delete_id(ctx: &ReducerContext, id: u64) {
-    ctx.db.person().id().delete(id)
-}
-```
-
-[macro library]: https://github.com/clockworklabs/SpacetimeDB/tree/master/crates/bindings-macro
-[module library]: https://github.com/clockworklabs/SpacetimeDB/tree/master/crates/lib
-[demo]: /#demo
+The Rust module SDK reference documentation is automatically generated from the module's source code and can be found [here](https://docs.rs/spacetimedb/latest/spacetimedb/).
```

Comment:
I'm not altogether opposed to using rustdoc for our docs, but I have a few concerns:

1. rustdoc is notoriously useless at documenting macros. We have a lot of macros. In particular, the current autogenerated docs for the `table` macro (which are actually entirely hand-written, and look relatively good all things considered) do not describe the generated table accessor interfaces at all.
2. If we are using rustdoc, at the very least we should host the docs ourselves, rather than just linking to docs.rs.
3. If this page is just going to have a link to rustdoc output, then it should just redirect to rustdoc. Why require the extra click?
4. Using rustdoc for the client SDK is much more challenging, as we also need to document the interfaces generated by the CLI. This is a solveable problem, but it is a problem that will need to be solved.
5. I would prefer to provide as consistent as possible a docs experience for all of C#, TypeScript and Rust. C# and TypeScript have rustdoc-like tools (I assume), but they will not look or feel consistent with each other, and none of them will look or feel like our website. There's an upside to this: Rust users are familiar with rustdoc, and presumably C# and TypeScript users are familiar with their ecosystem, so we can speak the reader's language, as it were. But this is something we'll have to consider more carefully.

Reply:
1. Perhaps we should document macros that are not properly documented in rustdoc on this page
2. What's the reason for this? Just better practice? We can certainly do it, just maybe want to save some time.
3. For #1
4. We can choose to use it for only the places we think it's appropriate. I don't disagree that it makes much less sense for the client SDK
5. I share your concerns on this one. It's a question of what the lesser of two evils is.

Reply:
> For https://github.com/clockworklabs/spacetime-docs/pull/1

I don't understand. What do you mean by this?

> We can choose to use it for only the places we think it's appropriate. I don't disagree that it makes much less sense for the client SDK

If we're going to use rustdoc, I think we should make an effort to use it for both the client SDK and the module library, for consistency's sake. I will need to discuss (with @RReverser and @jsdt probably?) whether this is feasible for the C# and TypeScript equivalents, but I have an idea in my head of how we can do it for Rust, which I will play around with and report back if it works.

Reply:
> What's the reason for this? Just better practice? We can certainly do it, just maybe want to save some time.

Self-hosting would allow us some marginal control over the style of our docs. I also think it looks more professional to have them on our own site. But this is not a big deal, and I'm happy to skip it in favor of using docs.rs because that's easier.